gene_finder.py

```diff
@@ -2,37 +2,48 @@
 """
 YOUR HEADER COMMENT HERE
 
-@author: YOUR NAME HERE
+@author: Aditya Kaushika
 
 """
 
 import random
 from amino_acids import aa, codons, aa_table   # you may find these useful
 from load import load_seq
 
+def count_v1(dna, base):
+    dna = list(dna)  # convert string to list of letters
+    i = 0            # counter
+    for c in dna:
+        if c == base:
+            i += 1
+    return i
+
 
 def shuffle_string(s):
     """Shuffles the characters in the input string
         NOTE: this is a helper function, you do not
         have to modify this in any way """
-    return ''.join(random.sample(s, len(s)))
+    return ''.join(random.sample(s, len(s))) # allows us to shuffle the letters in the code
 
 # YOU WILL START YOUR IMPLEMENTATION FROM HERE DOWN ###
 
-
 def get_complement(nucleotide):
-    """ Returns the complementary nucleotide
-
-        nucleotide: a nucleotide (A, C, G, or T) represented as a string
+    """ Returns the complementary nucleotide nucleotide: a nucleotide (A, C, G, or T) represented as a string
         returns: the complementary nucleotide
     >>> get_complement('A')
     'T'
     >>> get_complement('C')
     'G'
     """
-    # TODO: implement this
-    pass
-
+    for letter in nucleotide: #Go through every letter
+        if letter == 'A': #If Letter is A:
+            return 'T'      #Then complement it with T
+        elif letter == 'T': #Repeat with different letters.
+            return 'A'
+        elif letter == 'G':
+            return 'C'
+        else:
```

Comment:
this function will never fail if the user gives "A", "C', "G", "T"  only as inputs. But what if I give "S" as input?